### PR TITLE
regen

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
Part of https://github.com/gtk-rs/gir/pull/806

### CODE REVIEW

File: src/auto/attr_iterator.rs [hunk 1]
```diff
@@ -46,8 +46,8 @@ impl AttrIterator {
 
     pub fn range(&mut self) -> (i32, i32) {
         unsafe {
-            let mut start = mem::uninitialized();
-            let mut end = mem::uninitialized();
+            let mut start = mem::MaybeUninit::uninit().assume_init();
+            let mut end = mem::MaybeUninit::uninit().assume_init();
```
Comment (EPashkin) [inline src/auto/attr_iterator.rs]: In is not good example of "Incorrect usage of this method:" ? 
See https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#method.assume_init

Comment (GuillaumeGomez) [reply src/auto/attr_iterator.rs]: Yes and no. We want to have something in an uninitialised state as far as I understand it and we want it to be assumed as initialised so we can use it. But maybe I misunderstood the problem?

Comment (GuillaumeGomez) [reply src/auto/attr_iterator.rs]: The choice we have is to call `assume_init` *after* the C function was called.

Comment (EPashkin) [reply src/auto/attr_iterator.rs]: Seems correct usage is call after C function.
Also on `rustc 1.34.0` "MaybeUninit" is still "unstable feature" 😢

Comment (sdroege) [reply src/auto/attr_iterator.rs]: Yes the way how it is here now is undefined behaviour and defeats the purpose of `MaybeUninit`. You have to assert/assume that it's initialized after calling the C function.

You'd do
```rust
let mut foo = MaybeUninit::uninit();
ffi::foo_bar(foo.as_mut_ptr());
let foo = foo.assume_init();
```

That's defined behaviour and all good then.

Comment (sdroege) [reply src/auto/attr_iterator.rs]: >  Also on rustc 1.34.0 "MaybeUninit" is still "unstable feature" cry

It's stable in 1.36. As this is for the next gtk-rs release, updating the minimum required version to 1.36 seems fine to me.

Comment (GuillaumeGomez) [reply src/auto/attr_iterator.rs]: > ...
> That's defined behaviour and all good then.

So we all agree. I'll update the gir PR later on.

> It's stable in 1.36. As this is for the next gtk-rs release, updating the minimum required version to 1.36 seems fine to me.

Seems fine to me as well.

**Comment by GuillaumeGomez** [comment]
Updated.

File: src/auto/context.rs [hunk 2]
```diff
@@ -104,13 +104,16 @@ impl Context {
...
+            let mut n_families = mem::MaybeUninit::uninit();
             pango_sys::pango_context_list_families(
                 self.to_glib_none().0,
                 &mut families,
-                &mut n_families,
+                n_families.as_mut_ptr(),
             );
-            FromGlibContainer::from_glib_container_num(families, n_families as usize)
+            FromGlibContainer::from_glib_container_num(families, {
+                let n_families = n_families.assume_init();
```
Comment (sdroege) [inline src/auto/context.rs]: Why inside the call to `from_glib_container_num`? Not that it matters, seems just like this was more difficult to add to gir than the alternative?

**Comment by sdroege** [comment]
Looks good to me

File: src/auto/context.rs (hunk 2 above)
Comment (GuillaumeGomez) [reply src/auto/context.rs]: It's a special case. Normally we only need to do it after the function has been called, however, in `FromGlibContainer` and equivalents, it's done at the same time. This one is one of the rare that doesn't do both at once.

Comment (EPashkin) [reply src/auto/context.rs]: This also can changed to one-liner: https://github.com/gtk-rs/gir/pull/806#discussion_r300872790
so only difference from previous is `.assume_init()`

**Comment by GuillaumeGomez** [comment]
Updated correctly. :)

**Comment by EPashkin** [comment]
Thanks, 👍

**Comment by EPashkin** [comment]
.. after travis check on 1.36.0

**Comment by GuillaumeGomez** [comment]
Ah right, forgot to update travis build...

**Pull request merged**